core/pta/device.c

```diff
@@ -82,18 +86,31 @@ static TEE_Result invoke_command(void *pSessionContext __unused,
 				 uint32_t nCommandID, uint32_t nParamTypes,
 				 TEE_Param pParams[TEE_NUM_PARAMS])
 {
+	TEE_Result res = TEE_SUCCESS;
+	uint32_t rflags = 0;
+
 	switch (nCommandID) {
 	case PTA_CMD_GET_DEVICES:
-		return get_devices(nParamTypes, pParams,
-				   TA_FLAG_DEVICE_ENUM);
+		rflags = TA_FLAG_DEVICE_ENUM;
+		break;
 	case PTA_CMD_GET_DEVICES_SUPP:
-		return get_devices(nParamTypes, pParams,
-				   TA_FLAG_DEVICE_ENUM_SUPP);
-	default:
+		rflags = TA_FLAG_DEVICE_ENUM_SUPP;
+		if (IS_ENABLED(CFG_REE_FS))
+			rflags |= TA_FLAG_DEVICE_ENUM_SEC_STORAGE;
```

Comment:
- So here we have two flags set but `add_ta()` doesn't allow it and will return no TA
- What if `CFG_REE_FS` is disabled but `CFG_RPMB_FS` is enabled?

Reply:
> * So here we have two flags set but `add_ta()` doesn't allow it and will return no TA

`rflags` what's compared with `flags`, where the latter is from the TA and may only have one of the `TA_FLAG_DEVICE_ENUM`, `TA_FLAG_DEVICE_ENUM_SUPP`, `TA_FLAG_DEVICE_ENUM_SEC_STORAGE` flags set. `rflags` is permitted to have any bits set.

> * What if `CFG_REE_FS` is disabled but `CFG_RPMB_FS` is enabled?

Then will the `TA_FLAG_DEVICE_ENUM_SEC_STORAGE` bit be set during `PTA_CMD_GET_DEVICES_RPMB` instead.

Reply:
OK